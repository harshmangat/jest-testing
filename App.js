import { Text, View } from 'react-native'
import React, { Component } from 'react'

export class App extends Component {
  constructor(props) {
    super();
    this.state = {
      counter: 0,
    }
  }
  getData(x) {
  this.setState({counter: this.state.counter + x})
  }
  render() {
    return (
      <View>
        <Text>{this.state.counter}</Text>
      </View>
    )
  }
}

export default App