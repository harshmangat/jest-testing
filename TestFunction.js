const testFunction = x => {
  return x * 10;
};

const add = (a, b) => {
  return a + b;
};

const multiply = (a, b) => {
  return a * b;
};

const dataFetcher = async () => {
  const res = await fetch('https://jsonplaceholder.typicode.com/users');

  const data = await res.json();

  return data;
};
export {testFunction, add, multiply, dataFetcher};
