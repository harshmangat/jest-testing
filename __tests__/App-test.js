import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import {testFunction, add, multiply, dataFetcher} from '../TestFunction';

// test('renders correctly', () => {
//   const snapshot = renderer.create(<App />);
//   expect(snapshot).toMatchSnapshot();
// });

// test('function testing', () => {
//   const AppRef = renderer.create(<App />).getInstance();;
//   expect(AppRef.getData(2)).toEqual(12);
// });

// test('function testing', () => {
//   const AppRef = renderer.create(<App />).getInstance();
//   AppRef.getData(2);
//   expect(AppRef.state.counter).toEqual(2);
// });

describe('functionality of multiple functions', () => {
  test(' global function testing', () => {
    const input = testFunction(2);
    const output = 20;
    expect(input).toEqual(output);
  });
  test('sum', () => {
    const data = add(1, 2);
    expect(data).toEqual(3);
  });

  test('multiply', () => {
    const data = multiply(5, 5);
    expect(data).toEqual(25);
  });

  test('api testing', async () => {
    global.fetch = jest.fn().mockImplementation(() => {
      let data = new Promise((res, rej) => {
        res({
          json: function () {
            return {
              id: 1,

              name: 'Leanne Graham',

              username: 'Bret',

              email: 'Sincere@april.biz',

              phone: '1-770-736-8031 x56442',

              website: 'hildegard.org',
            };
          },
        });
      });

      return data;
    });

    const respo = await dataFetcher();

    expect(respo).toEqual({
      id: 1,

      name: 'Leanne Graham',

      username: 'Bret',

      email: 'Sincere@april.biz',

      phone: '1-770-736-8031 x56442',

      website: 'hildegard.org',
    });
  });
});
