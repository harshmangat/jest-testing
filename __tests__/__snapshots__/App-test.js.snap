// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functionality of multiple functions renders correctly 1`] = `
<View>
  <Text>
    0
  </Text>
</View>
`;
